### DISCUSSION TITLE
Where and how timers are used in the code of the game?

### DISCUSSION THREAD

**Discussion by EinstokFair** [singularity/singularity#313] [General]
We experience some random crashes (https://github.com/singularity/singularity/issues/306), that's why want to know this.

**Comment by nthykier** [comment]
Most likely the crashes are related to pygame/pygame#3128. At this point, the only known work around is to downgrade your pygame version.

As for how the timers are used - they are used to progress the game (i.e., they are active as soon as you unpause the game to determine when the game should advance the in-game time).  It is also involved in the world map day/night-cycle animation (but I think that happens indirectly as a side-effect of advancing the in-game time)
